addons/sale_management/models/sale_order_template.py

```diff
@@ -29,6 +29,7 @@ def _get_default_require_payment(self):
         domain=[('model', '=', 'sale.order')],
         help="This e-mail template will be sent on confirmation. Leave empty to send nothing.")
     active = fields.Boolean(default=True, help="If unchecked, it will allow you to hide the quotation template without removing it.")
+    company_id = fields.Many2one('res.company', string='Company')
```

Comment:
Maybe
```suggestion
    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id)
```

Reply:
in task, it is written that no need to set company by default

Reply:
@sswapnesh By default, products and templates will be shared and their company_id not set...

If someone wants to restrict access to a product or template, he can do so by setting the company_id himself.
We also make sure that we cannot set products from different companies on a template.

This doesn't avoid all buggy situations, but avoid most of them.  If the user plays with company-restricted products, and sets some of them on a quotation template, he will be expected to set the company_id to the company of the products.  If he doesn't, he will indeed see the template in other companies (and it will crash if he adds the template to quotations of other companies).  To avoid this case and still let the non restricted default, we would need to change the template company_id depending on its products, and it is a bit too "magical" in our opinion.

Reply:
Shouldn't we make this in align with `Common Product Catalog` ? (Would be much simple and aligned)
Cc; @dbo-odoo

Reply:
The _Common Product Catalog_ setting is removed with this PR.  Its behavior wasn't really clean from a technical point of view (disabling the rule on products access) and wouldn't work as well with new multi-company logic.

The default logic of sharing products will be kept, but when someone wants to limit products sharing, he will do so product by product, by setting the company_id.

Reply:
> Its behavior wasn't really clean from a technical point of view

Indeed, It is not but it was working well in Multi companies.  (And I think Common Contact Books is also working same logic)

> but when someone wants to limit products sharing, he will do so product by product, by setting the company_id.

So if company is set on Product , Limit visibility to that company only else They are shared among all. :+1: 

(Ps: Would be too much of work when someone wants to Add/Remove company, (Maybe a server action or something ))

Reply:
> (Ps: Would be too much of work when someone wants to Add/Remove company, (Maybe a server action or something ))

We could consider adding an action for products (and other models?) company migration.  
Cc: @dbo-odoo ?